### ISSUE TITLE
Injecting restify server

### ISSUE THREAD

**Issue by royibernthal** [pleerock/typedi#5]
Hi,

I'd like to inject the Server created by restify.createServer().

It should look something like this:
`Container.set(restify.Server, server);`

I'm providing the instance once the createServer() promise is resolved successfully.

The problem is, Server is an interface and not a class, it is not described as a class anywhere in restify's definition file.

Naturally, I'm able to access the Server instance's prototype, but it is not described via typescript anywhere.

How would you suggest I should inject it?

Even if providing the instance as follows would work (I don't know if it would):
`Container.set((server as any).prototype, server);`

I won't be able to do something like this in the class I'm injecting into (since here Server would be referring to an interface):
```
@Inject()
server: Server;
```

Would named services be the solution? Is there a way to do it without them?

**Comment by pleerock** [comment]
if server is an interface you should use [named services](https://github.com/pleerock/typedi#named-services) and set it like this:

`Container.set("name-of-your-service", server);`

**Comment by royibernthal** [comment]
These are the signatures from Container.d.ts:

```
static set(type: Function, value: any): void;
static set(name: string, type: Function, value: any): void;
```

Is there supposed to be one like that?

`static set(type: string, value: any): void;`

**Comment by pleerock** [comment]
this seems to be incorrect. Fixed in `0.4.3` please try that out.

**Comment by royibernthal** [comment]
I tried the new version, it worked as you said. Thank you very much.
